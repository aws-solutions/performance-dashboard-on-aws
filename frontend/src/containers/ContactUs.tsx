/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import MarkdownRender from "../components/MarkdownRender";
import { usePublicSettings } from "../hooks";
import { useTranslation } from "react-i18next";
import Breadcrumbs from "../components/Breadcrumbs";
import { faEnvelope } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import Spinner from "../components/Spinner";

function ContactUs() {
    const { t } = useTranslation();
    const { settings, loadingSettings } = usePublicSettings(true);

    return (
        <>
            <Breadcrumbs
                crumbs={[
                    {
                        label: t("Home"),
                        url: "/",
                    },
                    {
                        label: t("ContactUs"),
                    },
                ]}
            />

            <h1 id="contactUsLabel">{t("ContactUs")}</h1>

            {loadingSettings ? (
                <Spinner className="text-center margin-top-9" label={t("LoadingSpinnerLabel")} />
            ) : (
                <div>
                    <MarkdownRender
                        source={settings.contactUsContent}
                        className="margin-bottom-2"
                    />

                    <div className="grid-row">
                        <div className="grid-col flex-9">
                            <div className="published-site font-sans-lg">
                                {settings.contactEmailAddress && (
                                    <div>
                                        {settings.contactEmailAddress}
                                        <a
                                            title={t("ContactUs")}
                                            aria-label={t("ContactUs")}
                                            href={`mailto:${
                                                settings.contactEmailAddress
                                            }?subject=${t(
                                                "Public.PerformanceDashboardAssistance",
                                            )}`}
                                            className="usa-link margin-left-1"
                                        >
                                            <FontAwesomeIcon
                                                size="1x"
                                                icon={faEnvelope}
                                                className="margin-right-1"
                                            />
                                        </a>
                                    </div>
                                )}
                            </div>
                        </div>
                        <div className="grid-col flex-3 text-right"></div>
                    </div>
                </div>
            )}
        </>
    );
}

export default ContactUs;
