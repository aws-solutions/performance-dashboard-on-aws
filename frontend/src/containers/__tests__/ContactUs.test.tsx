/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import ContactUs from "../ContactUs";

test("renders the component", async () => {
    const wrapper = render(<ContactUs />, { wrapper: MemoryRouter });
    expect(wrapper.container).toMatchSnapshot();
});
