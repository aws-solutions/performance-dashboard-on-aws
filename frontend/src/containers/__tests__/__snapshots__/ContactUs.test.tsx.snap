// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<div>
  <div
    class="padding-1"
  >
    <nav
      aria-label="Breadcrumbs"
      class="usa-breadcrumb usa-breadcrumb--wrap"
    >
      <ol
        class="usa-breadcrumb__list"
      >
        <li
          class="usa-breadcrumb__list-item"
        >
          <a
            class="usa-breadcrumb__link"
            href="/"
          >
            <span>
              Home
            </span>
          </a>
        </li>
        <li
          aria-current="page"
          class="usa-breadcrumb__list-item usa-current"
        >
          <span>
            Contact Us
          </span>
        </li>
      </ol>
    </nav>
  </div>
  <h1
    id="contactUsLabel"
  >
    Contact Us
  </h1>
  <div>
    <div
      class="Markdown margin-bottom-2"
    />
    <div
      class="grid-row"
    >
      <div
        class="grid-col flex-9"
      >
        <div
          class="published-site font-sans-lg"
        />
      </div>
      <div
        class="grid-col flex-3 text-right"
      />
    </div>
  </div>
</div>
`;
